### DISCUSSION TITLE
Reading russian text from file throws an error

### DISCUSSION THREAD

**Discussion by thewolfram** [nlohmann/json#3198] [Q&A] [answered]
Hi, I'm trying to read russian text from file but it gives me strange error when debugging and program crashes. I'm using Windows 10, Visual Studio 2022 and latest release of json.hpp

```cpp
int main()
{
    setlocale(LC_ALL, "Russian");
    json j;

    std::ifstream f("file.txt");
    j = nlohmann::json::parse(f); //crash here
    f.close();

    std::cout << j.dump() << std::endl;

    std::cin.get();

    return 0;
}
```

![1](https://user-images.githubusercontent.com/66202986/146438375-e0e8d541-38fb-4550-bf98-674e6a44ebe7.png)

**Comment by nlohmann** [comment]
Please make sure your file is UTF-8 encoded. Please also check the exception message which should indicate the location of the parse error.

**Reply by thewolfram** [reply]
> 

It doesn't crash now, but I get strange symbols instead of cyrillic letters, word "тест" looks like this "С'РчС?С'"
![image](https://user-images.githubusercontent.com/66202986/146450989-4e7dd3ea-ebea-4fdd-8120-a17bde01e854.png)

**Comment by thewolfram** [comment, marked as answer]
So when I save file as ANSI, I can get russian text with ifstream without any issues, but then json will cause crash, If I change file to utf-8 then ifstream will give me the same json gives.

**Reply by nlohmann** [reply]
This library only supports UTF-8. If the read strings are displayed weird, please make sure your output is interpreted as UTF-8.

If you still have issues, please provide more information.

**Reply by thewolfram** [reply]
> This library only supports UTF-8. If the read strings are displayed weird, please make sure your output is interpreted as UTF-8.
> 
> If you still have issues, please provide more information.

Thank you very much I added SetConsoleOutputCP(CP_UTF8); at top of main() and everything displays correctly now!
